test/dhcp/dhcp_api_test.rb

```diff
@@ -316,34 +335,40 @@ def test_delete_record
 
   def test_delete_records_by_ip
     @server.expects(:del_records_by_ip).with("192.168.122.0", "192.168.122.1")
+    @server.stubs(:validate_supported_address).returns(true)
     delete "/192.168.122.0/ip/192.168.122.1"
     assert_equal 200, last_response.status
     assert_empty last_response.body
   end
 
   def test_delete_records_by_ip_for_nonexistent_subnet
     @server.expects(:del_records_by_ip).with("192.168.122.0", "192.168.122.1").raises(::Proxy::DHCP::SubnetNotFound)
+    @server.stubs(:validate_supported_address).returns(true)
     delete "/192.168.122.0/ip/192.168.122.1"
     assert_equal 200, last_response.status
     assert_empty last_response.body
   end
 
   def test_delete_records_by_mac
     @server.expects(:del_record_by_mac).with("192.168.122.0", "00:11:bb:cc:dd:ee")
+    @server.stubs(:validate_supported_address).returns(true)
     delete "/192.168.122.0/mac/00:11:bb:cc:dd:ee"
     assert_equal 200, last_response.status
     assert_empty last_response.body
   end
 
   def test_delete_records_by_mac_uppercase
     @server.expects(:del_record_by_mac).with("192.168.122.0", "00:11:bb:cc:dd:ee")
+    @server.stubs(:validate_supported_address).returns(true)
     delete "/192.168.122.0/mac/00:11:BB:CC:DD:EE"
     assert_equal 200, last_response.status
     assert_empty last_response.body
   end
 
   def test_delete_records_by_mac_for_nonexistent_subnet
     @server.expects(:del_record_by_mac).with("192.168.122.0", "00:11:bb:cc:dd:ee").raises(::Proxy::DHCP::SubnetNotFound)
+    @server.stubs(:validate_supported_address).returns(true)
```

Comment:
Apologies for another comment, I should have see it.

Can you move all these statements into the `setup` method above? It's being called for each test and it is exactly what intended for. You can override `stubs` with `expects` in any test if you want to test for it.